clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -367,19 +413,25 @@ private void pollSelectionKeys(Iterable<SelectionKey> selectionKeys,
                                 socketChannel.socket().getSendBufferSize(),
                                 socketChannel.socket().getSoTimeout(),
                                 channel.id());
+                        madeReadProgressLastPoll = true;
```

Comment:
i was being cautious. but youre right, given the progress flag is only used in combination with dataInBuffers its probably impossible for a poll() round to involving the progress flag to consist solely of handshaking and connecting operations.